reports/tasks.py

```diff
@@ -364,6 +365,19 @@ def handle_sms_delivery_msisdn(
         for outbound in outbounds:
             if 'voice_speech_url' not in outbound.get('metadata', {}):
 
+                if (not outbound.get('to_addr', '') and
+                        outbound.get('to_identity', '')):
+                    identity = self.get_identity(
+                        ids_client, outbound['to_identity'])
+
+                    details = identity.get('details', {})
+                    default_addr_type = details.get(
+                        'default_addr_type', 'msisdn')
+
+                    addresses = details.get('addresses', {})
+                    outbound['to_addr'] = list(
+                        addresses.get(default_addr_type, {}).keys())[0]
```

Comment:
some msisdns have the `default` key so we should probably prefer those over others